_studio/mfx_lib/decode/mpeg2/src/mfx_mpeg2_decode.cpp

```diff
@@ -651,31 +651,36 @@ mfxStatus VideoDECODEMPEG2::DecodeFrameCheck(mfxBitstream* bs, mfxFrameSurface1*
     MFX_CHECK(m_core, MFX_ERR_UNDEFINED_BEHAVIOR);
     MFX_CHECK(m_decoder, MFX_ERR_NOT_INITIALIZED);
 
-    MFX_CHECK_NULL_PTR2(surface_work, surface_out);
+    MFX_CHECK_NULL_PTR1(surface_out);
 
     mfxStatus sts = bs ? CheckBitstream(bs) : MFX_ERR_NONE;
```

Comment:
I don't like this construction
It should be
```
mfxStatus sts = MFX_ERR_NONE;
if (bs)
{
    sts = CheckBitstream(bs);
    MFX_CHECK_STS(sts);
}
```

But since it wasn't introduced by you I'm not forcing you to change it